src/Package/Impl/Utilities/ProjectUtilities.cs

```diff
@@ -42,5 +52,95 @@ public static class ProjectUtilities {
 
             return result;
         }
+
+        public static EnvDTE.Project GetActiveProject() {
+            DTE dte = VsAppShell.Current.GetGlobalService<DTE>();
+            var projects = (Array)dte.ActiveSolutionProjects;
+            if (projects != null && projects.Length == 1) {
+                return (EnvDTE.Project)projects.GetValue(0);
+            }
+            return null;
+        }
+
+        public static void AddNewItem(string templateName, string name, string extension, string destinationPath) {
```

Comment:
Actually, looking at what C# and C++ do, it doesn't just add the item right away when you click "Add Class". Instead, it opens the usual "Add New Item" dialog (which lets the user provide the name for the class), but pre-selects the class template. I assume other "Add ..." commands work in a similar fashion.

Both C# and C++ use `IVsAddProjectItemDlg.AddProjectItemDlg` for this purpose - there is a parameter there to pre-select a template.